providers/implementations/macs/kmac_prov.c

```diff
@@ -422,6 +450,13 @@ static int kmac_set_ctx_params(void *vmacctx, const OSSL_PARAM *params)
             ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_OUTPUT_LENGTH);
             return 0;
         }
+#ifdef FIPS_MODULE
+        if (FIPS_kmac_length_checks_enabled(PROV_LIBCTX_OF(kctx->provctx))
+                && kctx->out_len < KMAC_MIN_OUTPUT_LEN_FIPS) {
```

Comment:
This check seems to occur on the old `kctx->out_len`, i.e. before we have updated it??

Reply:
Good catch.